File: DSA/DSA06024 SẮP XẾP CHỌN.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A) cin >> i;

    for (int i = 0; i < n-1; i++)
    {
        int mini = i;
        for (int j = i+1; j < n; j++)
        {
            if (A[j] < A[mini])
            {
                mini = j;
            }
        }
        swap(A[mini], A[i]);

        cout << "Buoc " << i+1 << ": ";
        for (auto &&k : A) cout << k << " ";
        cout << endl;
    }
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    submit();

    return 0;
}

File: DSA_python/DSA06024 SẮP XẾP CHỌN.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))
    
    for i in range(n-1):
        mini = i
        for j in range(i+1, n):
            if A[j] < A[mini]:
                mini = j
        A[mini], A[i] = A[i], A[mini]

        print(f"Buoc {i+1}:", " ".join(map(str, A)))


submit()
